### PR TITLE
Additions to DenseMatrixBase and MutableDenseMatrix

### PR DESCRIPTION
**Pull request by ShikharJ**
Relevant: #17

### CODE REVIEW

**Comment by ShikharJ** [comment]
@isuruf Though I have updated the `symengine_version.txt` to the latest commit, the above methods are still not visible in `SymEngine.py`. Please advise.

**Comment by isuruf** [comment]
It's using the correct version of symengine. Something is wrong with your code here.

File: symengine/lib/symengine.pxd [hunk 1]
```diff
@@ -671,6 +671,14 @@ cdef extern from "<symengine/matrix.h>" namespace "SymEngine":
             const DenseMatrix &x, DenseMatrix &result) nogil except +
     void diff "SymEngine::sdiff"(const DenseMatrix &A,
             RCP[const Basic] &x, DenseMatrix &result) nogil except +
+    void row_join(const DenseMatrix &A, const DenseMatrix &B, DenseMatrix &C) nogil
```
Comment (isuruf) [inline symengine/lib/symengine.pxd]: This should be declared in the class itself without the first parameter. See `add_matrix`

Comment (ShikharJ) [inline symengine/lib/symengine.pxd]: But `add_matrix` by default has only two parameters. Changing the declarations similarly has no effect as well.

**Comment by ShikharJ** [comment]
@isuruf There doesn't seem to be any effect with changing the declarations. What do you suggest we do?

**Comment by ShikharJ** [comment]
Ping @isuruf

**Comment by ShikharJ** [comment]
@isuruf Can yoy also take a look here. I'd like to finish this up before the release.

**Comment by isuruf** [comment]
This can't go in the release, since it's using functionality of symengine not yet released. I'll have a look about what can do.

**Comment by ShikharJ** [comment]
@isuruf Anything on this? I'd like to finish this up.

**Comment by ShikharJ** [comment]
Ping @isuruf.

**Comment by isuruf** [comment]
https://github.com/symengine/symengine/pull/1269/files#r128920801

File: symengine/lib/symengine_wrapper.pyx [hunk 2]
```diff
@@ -2826,6 +2842,8 @@ cdef class DenseMatrixBase(MatrixBase):
     def tolist(self):
         return self[:]
 
+    _mat = tolist
```
Comment (isuruf) [inline symengine/lib/symengine_wrapper.pyx]: This is really inefficient. `_mat` method should just return `self`.

File: symengine/lib/symengine.pxd [hunk 3]
```diff
@@ -769,6 +769,18 @@ cdef extern from "<symengine/matrix.h>" namespace "SymEngine":
...
+        void row_insert(const DenseMatrix &B, unsigned pos) nogil
+        void col_insert(const DenseMatrix &B, unsigned pos) nogil
+        void row_del(unsigned k) nogil
+        void col_del(unsigned k) nogil
+        void row_exchange_dense(DenseMatrix &A, unsigned i, unsigned j) nogil
+        void row_mul_scalar_dense(DenseMatrix &A, unsigned i, RCP[const Basic] &c) nogil
+        void row_add_row_dense(DenseMatrix &A, unsigned i, unsigned j, RCP[const Basic] &c) nogil
+        void column_exchange_dense(DenseMatrix &A, unsigned i, unsigned j) nogil
+        void dot(const DenseMatrix &A, const DenseMatrix &B, DenseMatrix &C) nogil
+        void cross(const DenseMatrix &A, const DenseMatrix &B, DenseMatrix &C) nogil
```
Comment (ShikharJ) [inline symengine/lib/symengine.pxd]: @isuruf Can you point out what is wrong with the declarations here?

Comment (isuruf) [inline symengine/lib/symengine.pxd]: They are not declared in class `DenseMatrix`. They should be declared outside the class.

**Comment by ShikharJ** [comment]
Can you review this once @isuruf?

**Comment by isuruf** [comment]
You need to declare the function outside the class in symengine.
See https://github.com/symengine/symengine/blob/master/symengine/matrix.h#L226 and https://github.com/symengine/symengine/blob/master/symengine/matrix.h#L401

**Comment by ShikharJ** [comment]
@isuruf The tests pass now, can you review this please?

**Comment by ShikharJ** [comment]
Ping @isuruf.

**Pull request merged**